### DISCUSSION TITLE
Benefits of contributing to an OSS project

### DISCUSSION THREAD

**Discussion by reubenalfred** [Megabit/Blazorise#4837] [General]
If you are a hobbyist programmer or a recent graduate and looking for ways to learn new skills, exposure to different technologies and tools then contributing to an  OSS (open source software) project is, IMHO, the best place to start from.

One can contribute in so many ways and wide and varying skills are required. Contribution in coding, documentation, graphic design, marketing, legal, research, web development are required all the time and the level of proficiency won't be an issue as long as one has a genuine desire to learn new skills then the proficiency will improve.

Some individuals have the desire to contribute but lack the confidence in asking questions or making mistakes or being corrected or might even think that they don't have the time to contribute. OSS community only look for willing individuals, nobody is perfect.

From my comments or style of writing you may have noticed that I am pretty average or even less but all I have is the willing spirit 😉

A smart person knows how to turn junk into treasure and in OSS projects there are very smart people around who can turn you into a trove of treasure.

1 hour a week to contribute is a good way to start to familiarise oneself to how things are done in a project.
If someone gets stuck then questions can be posted in the forum and if you are stuck then you find something else to do in the project for the rest of the time slot until you get a response. Eventually you'll find ways to get answers quickly to problems.

So the following are the skills that we can develop when we regularly contribute to an OSS project.

- learn something new or improve on what you already know
- for new graduates this is valuable experience to your resume
- learn to work on your own
- study professionally written code to see how software engineering principles are applied
- improve problem solving skills

And many many more others will be adding more below...

**Comment by stsrki** [comment]
Contacting people from our Discord server and mentioning this discussion might be good. I don't think many people actively look into GitHub for discussion threads.

**Reply by reubenalfred** [reply]
Yup will sort that out...
